frontend/app/src/main/java/dk/scheduling/schedulingfrontend/pages/CreateTaskPage.kt

```diff
@@ -0,0 +1,221 @@
+            onSelect = { selectedItem = it },
+        )
+
+        // Duration input field
+        Spacer(modifier = Modifier.height(20.dp))
+
+        var duration: String by rememberSaveable { mutableStateOf("") }
+        var numbersOnly: Boolean by rememberSaveable { mutableStateOf(true) }
+
+        OutlinedTextField(
+            modifier = modifier.fillMaxWidth(),
+            value = duration,
+            onValueChange = {
+                duration = it
+                numbersOnly = numbersOnly(duration)
+            },
+            label = { Text("Duration (m)") },
+            singleLine = true,
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+            isError = !numbersOnly,
+        )
+
+        Spacer(modifier = Modifier.height(20.dp))
+
+        // start date & end date
+        val dateRangeDialog = remember { mutableStateOf(false) }
+        val datePickerValue = remember { mutableStateOf(DateRange()) }
+
+        StandardDateRangePicker(
+            closeDialog = { dateRangeDialog.value = false },
+            passingDate = { datePickerValue.value = it },
+            openDialog = dateRangeDialog.value,
+        )
+
+        val dateMsg: String = if (datePickerValue.value.isValidRange()) "no interval selected." else datePickerValue.value.print()
+
+        ClickableCard({ dateRangeDialog.value = true }, "Date interval: $dateMsg")
+
+        Spacer(modifier = Modifier.height(30.dp))
+
+        // start time
+        val startTimeDialog = remember { mutableStateOf(false) }
+        val startTimeState = rememberTimePickerState()
+
+        StandardTimePickerDialog(
+            closeDialog = { startTimeDialog.value = false },
+            state = startTimeState,
+            openDialog = startTimeDialog.value,
+        )
+        ClickableCard({ startTimeDialog.value = true }, "Start time: ${formatTime(startTimeState.hour, startTimeState.minute)}")
+
+        // end time
+        Spacer(modifier = Modifier.height(30.dp))
+
+        val endTimeDialog = remember { mutableStateOf(false) }
+        val endTimeState = rememberTimePickerState()
+
+        StandardTimePickerDialog(
+            closeDialog = { endTimeDialog.value = false },
+            state = endTimeState,
+            openDialog = endTimeDialog.value,
+        )
+        ClickableCard({ endTimeDialog.value = true }, "End time: ${formatTime(endTimeState.hour, endTimeState.minute)}")
+
+        // Submit or Cancel
+        Spacer(modifier = Modifier.height(30.dp))
+
+        Button(
+            onClick = { handleSubmission() },
+            enabled = isValidInput(duration, datePickerValue.value, startTimeState, endTimeState),
+            modifier = modifier.fillMaxWidth(),
+        ) {
+            Text(text = "Create task")
+        }
+
+        FilledButton(
+            onClick = {
+                handleCancellation()
+            },
+            text = "Cancel",
+        )
+    }
+}
+
+@Composable
+fun ClickableCard(
```

Comment:
It seems a little strange for this to be cards. Why not buttons?

Reply:
So it seems more like fields and less like buttons? We usually dont print info in the field of a button?